### PR TITLE
Bake seed with data

### PR DESCRIPTION
**Pull request by andrej-griniuk**
Add the ability to bake seeds containing data selected form the target table.

### CODE REVIEW

**Comment by lorenzo** [comment]
This looks good to me, thanks for the contribution!

Do you mind adding a couple tests with data that could confuse your prettifyArray? Like a column having `array(asdas)` or similar

File: src/Shell/Task/SeedTask.php [hunk 1]
```diff
@@ -84,10 +85,33 @@ public function templateData()
             $table = $this->params['table'];
         }
 
+        $records = false;
+        if ($this->param('data')) {
+            $limit = (int)$this->param('limit');
+
+            $fields = $this->param('fields') ?: '*';
+            if ($fields != '*') {
```
Comment (dereuromark) [inline src/Shell/Task/SeedTask.php]: All String comparisons should be strict.

File: src/Shell/Task/SeedTask.php [hunk 2]
```diff
@@ -140,8 +164,90 @@ public function getOptionParser()
...
+    protected function prettifyArray($array, $tabCount = 3, $indentCharacter = "    ")
+    {
+        $content = var_export($array, true);
+
+        $lines = explode("\n", $content);
+
+        $inString = false;
+
+        foreach ($lines as $k => &$line) {
+            if ($k == 0) {
```
Comment (dereuromark) [inline src/Shell/Task/SeedTask.php]: those two can also be strict, as the key must be numeric (coming from explode function).

File: src/Shell/Task/SeedTask.php [hunk 3]
```diff
@@ -140,8 +164,90 @@ public function getOptionParser()
...
+                // First row
+                $line = '[';
+                continue;
+            } elseif ($k == count($lines) - 1) {
```
Comment (dereuromark) [inline src/Shell/Task/SeedTask.php]: no else required here, just an if in a new line.

File: tests/comparisons/Seeds/testWithDataAndLimit.php [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+                'title' => 'Lorem ipsum dolor sit amet',
+                'description' => 'Lorem ipsum dolor sit amet, aliquet feugiat.',
+                'published' => 'Y',
+            ],
+            [
+                'id' => '2',
+                'title' => 'Second event',
+                'description' => 'Second event description.',
+                'published' => 'Y',
+            ],
```
Comment (inoas) [inline tests/comparisons/Seeds/testWithDataAndLimit.php]: Do we have a code guide on array tail commas?

File: tests/comparisons/Seeds/testWithDataAndFields.php [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+ * Events seed.
+ */
+class EventsSeed extends AbstractSeed
+{
+    /**
+     * Run Method.
+     *
+     * Write your database seeder using this method.
+     *
+     * More information on writing seeders is available here:
```
Comment (inoas) [inline tests/comparisons/Seeds/testWithDataAndFields.php]: Seeds not Seeders, as far as I know.

File: .gitignore [hunk 6]
```diff
@@ -1,2 +1,3 @@
 plugins/
 vendor/
+.idea
```
Comment (inoas) [inline .gitignore]: Seems unrelated. Not sure it should be there by default.

Comment (lorenzo) [inline .gitignore]: It's ok to add it

File: tests/comparisons/Seeds/testWithDataAndLimit.php (hunk 4 above)
Comment (lorenzo) [inline tests/comparisons/Seeds/testWithDataAndLimit.php]: I'm not sure we do, but I like those

Comment (andrej-griniuk) [inline tests/comparisons/Seeds/testWithDataAndLimit.php]: Well, it passes through phpcs and var_export does add them, so don't think it's worth bothering with removing them.

File: .gitignore (hunk 6 above)
Comment (andrej-griniuk) [inline .gitignore]: I would add even more there, at least composer.lock. Also when you run tests they generate another bunch of files that are asking to be commited.

File: tests/comparisons/Seeds/testWithDataAndLimit.php (hunk 4 above)
Comment (inoas) [inline tests/comparisons/Seeds/testWithDataAndLimit.php]: @lorenzo I like them too, but cake core omits them as far as I know.
I'd be great to have an official stance on this.

Comment (dereuromark) [inline tests/comparisons/Seeds/testWithDataAndLimit.php]: Adding a sniff for it should be trivial.

File: .gitignore (hunk 6 above)
Comment (lorenzo) [inline .gitignore]: composer.lock should be in the repo

File: tests/comparisons/Seeds/testWithDataAndLimit.php (hunk 4 above)
Comment (andrej-griniuk) [inline tests/comparisons/Seeds/testWithDataAndLimit.php]: cakephp/cakephp-codesniffer doesn't mind tail commas either, so don't see any issues here.

File: .gitignore (hunk 6 above)
Comment (jadb) [inline .gitignore]: `.idea` should be added at a developer's global level (`~/.gitignore`) not in every plugin's `.gitignore` IMO.

Comment (andrej-griniuk) [inline .gitignore]: Ok, I'll just roll it back - irrelevant to this PR anyway.

File: tests/comparisons/Seeds/testWithDataAndLimit.php (hunk 4 above)
Comment (inoas) [inline tests/comparisons/Seeds/testWithDataAndLimit.php]: @andrej-griniuk Not a fault with your PR at all. I was just wondering if we can stick to one consistent style.

File: src/Shell/Task/SeedTask.php [hunk 7]
```diff
@@ -140,8 +164,92 @@ public function getOptionParser()
             'choices' => $bakeThemes
         ])->addArgument('name', [
             'help' => 'Name of the seed to bake. Can use Plugin.name to bake plugin models.'
+        ])->addOption('data', [
+            'short' => 'd',
+            'boolean' => true,
+            'help' => 'Include data from the table to the seed'
+        ])->addOption('fields', [
+            'short' => 'f',
```
Comment (lorenzo) [inline src/Shell/Task/SeedTask.php]: `f` is already used for `force` I suggest having no short form for this and also remove it for `--data`

**Comment by andrej-griniuk** [comment]
Hey guys,
so what's the status here?

**Comment by lorenzo** [comment]
Sorry for the delay @andrej-griniuk and thanks for the good work!

**Comment by HavokInspiration** [comment]
Refs #191
